### PR TITLE
Add chest module

### PR DESCRIPTION
**Pull request by curio-sitas**
# Feature
This PR is about adding a new module to the scanner, the chest module. It is able to scan chests (normal, ender and trapped), barrels and shulker boxes. The feature has been added (and tested) on both fabric and forge implementations.

Here are some screenshots : 
# Recipe

# Scan 

# Todo
- [ ] Translate in other language than english

### CODE REVIEW

File: common/src/main/java/li/cil/scannable/common/config/CommonConfig.java [hunk 1]
```diff
@@ -118,4 +122,19 @@ public final class CommonConfig {
...
+
+    @ExpectPlatform
+    private static Set<ResourceLocation> getDefaultChestsTags() {
+        throw new AssertionError();
+    }
+
+    @Path("chests") @WorldRestart
+    @Comment("Registry names of blocks considered 'chests', requiring the chest scanner module.")
+    @ItemType(ResourceLocation.class)
+    public static Set<ResourceLocation> commonChestTags = getDefaultChestsTags();
```
Comment (fnuecke) [inline common/src/main/java/li/cil/scannable/common/config/CommonConfig.java]: Please stick to the order of things in this class, i.e. fields first, then methods.
So have the new fields above the `getDefaultCommonOreTags()`.

File: common/src/main/resources/assets/scannable/lang/en_us.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
   "item.scannable.entity_module": "Scanner Module: Living Entity",
   "item.scannable.entity_module.desc": "Detects configured living entities. Use on an entity to add it to the configuration. Use on air to manage configuration",
   "item.scannable.entity_module.list": "Configured entities:",
-
+  "item.scannable.chest_module": "Scanner Module: Chests",
+  "item.scannable.chest_module.desc": "Detects chests blocks (e.g. chests, barrels, shulker boxes).",
```
Comment (fnuecke) [inline common/src/main/resources/assets/scannable/lang/en_us.json]: Think this should read "Detects chest blocks" (i.e. no plural on the chest), no?  That said, given the list it picks up, a better description might be "Detects storage blocks"?

File: fabric/src/main/resources/data/scannable/recipes/chest_module.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+            "item": "scannable:blank_module"
+        },
+        {
+            "item": "minecraft:chest"
+        }
+    ],
+    "result": {
+        "item": "scannable:chest_module"
+    }
+}
```
Comment (fnuecke) [inline fabric/src/main/resources/data/scannable/recipes/chest_module.json]: Please stick to formatting of other data files for consistency (using 2 space indentation, which IIRC is also what pops out when running the data generators).

File: forge/src/main/resources/data/scannable/recipes/chest_module.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "type": "minecraft:crafting_shapeless",
+  "category": "misc",
+  "group": "scanner_module",
+    "ingredients": [
+        {
+            "item": "scannable:blank_module"
+        },
+        {
+            "item": "minecraft:chest"
+        }
```
Comment (fnuecke) [inline forge/src/main/resources/data/scannable/recipes/chest_module.json]: Please stick to consistent formatting here, too (2 space indentation).

File: common/src/main/java/li/cil/scannable/common/scanning/ChestScannerModule.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+public enum ChestScannerModule implements BlockScannerModule {
+    INSTANCE;
+
+    private Predicate<BlockState> filter;
+
+    public static void clearCache() {
```
Comment (fnuecke) [inline common/src/main/java/li/cil/scannable/common/scanning/ChestScannerModule.java:27]: I don't see this being called from the `ProviderCacheManagerImpl` classes?

**Review by fnuecke — CHANGES_REQUESTED** [review]
Thanks for your work on this!  Just a couple minor and one functional remarks.

**Comment by curio-sitas** [comment]
Sorry for the formating mistakes, i resolved all your remarks.

File: common/src/main/java/li/cil/scannable/common/scanning/ChestScannerModule.java (hunk 5 above)
Comment (fnuecke) [reply common/src/main/java/li/cil/scannable/common/scanning/ChestScannerModule.java:27]: Ah, sorry, I was not clear enough here: I did not mean to the method can be removed, but that it should be called by these classes - as is the case for the modules that cache their filters, like the common/rare ores ones.

**Comment by fnuecke** [comment]
Also to make the automated checks happy, you'll need to run `./gradlew :spotlessApply` or clean up manually (see workflow output).

**Comment by curio-sitas** [comment]
> Also to make the automated checks happy, you'll need to run `./gradlew :spotlessApply` or clean up manually (see workflow output).

I used the command before push the commit

**Comment by fnuecke** [comment]
If I'm not mistaken the `clearCache` call is still missing from the Forge `ProviderCacheManagerImpl` (I'm only seeing it in the Fabric one)?

Once that's in I think this one's good to go :)

**Comment by curio-sitas** [comment]
> If I'm not mistaken the `clearCache` call is still missing from the Forge `ProviderCacheManagerImpl` (I'm only seeing it in the Fabric one)?
> 
> Once that's in I think this one's good to go :)

Done 👍 
Thank you for your review

**Review by fnuecke — APPROVED** [review]

**Comment by fnuecke** [comment]
Thanks for the PR!

**Pull request merged**